### PR TITLE
[RNMobile] Update `react-native-reanimated` version to `2.4.1`

### PR DESCRIPTION
**Pull request by fluiddot**
**Gutenberg Mobile PR:** https://github.com/wordpress-mobile/gutenberg-mobile/pull/4653
**WordPress-Android PR:** https://github.com/wordpress-mobile/WordPress-Android/pull/16106
**WordPress-iOS PR:** https://github.com/wordpress-mobile/WordPress-iOS/pull/18137

## What?

Updates `react-native-reanimated` (i.e. Reanimated library) package to version `2.4.1`.

## Why?

The current version of Reanimated doesn't allow debugging with the Chrome debugger. However, in newer versions this has been addressed hence, updating the package will allow us to re-enable the Chrome debugger. 

## How?

[A PR](https://github.com/wordpress-mobile/react-native-reanimated/pull/13) has been created in the forked repository of the package, and the version has been bumped in the `package.json` file.

## Testing Instructions

1. Apply the following patch diff to incorporate an example that uses Reanimated:
<details><summary>Click here to display the patch diff</summary>

```diff
diff --git a/packages/edit-post/src/components/visual-editor/index.native.js b/packages/edit-post/src/components/visual-editor/index.native.js
index 886e56e493938b4b63c08199e39fc90f7accf4e5..25850c6b5cab4e0f2808064fdbd79c7c2481c0d7 100644
--- a/packages/edit-post/src/components/visual-editor/index.native.js
+++ b/packages/edit-post/src/components/visual-editor/index.native.js
@@ -6,12 +6,77 @@ import { BlockList } from '@wordpress/block-editor';
 /**
  * External dependencies
  */
-import { Keyboard } from 'react-native';
+import { Keyboard, StyleSheet, View } from 'react-native';
+import { Gesture, GestureDetector } from 'react-native-gesture-handler';
+import Animated, {
+	useSharedValue,
+	useAnimatedStyle,
+	withSpring,
+} from 'react-native-reanimated';
 /**
  * Internal dependencies
  */
 import Header from './header';
 
+const TestComponent = () => {
+	const ballStyles = StyleSheet.create( {
+		container: {
+			position: 'absolute',
+			bottom: 64,
+			right: 32,
+			zIndex: 10,
+		},
+		ball: {
+			width: 100,
+			height: 100,
+			borderRadius: 100,
+			backgroundColor: 'blue',
+		},
+	} );
+
+	const isPressed = useSharedValue( false );
+	const offset = useSharedValue( { x: 0, y: 0 } );
+	const animatedStyles = useAnimatedStyle( () => {
+		return {
+			transform: [
+				{ translateX: offset.value.x },
+				{ translateY: offset.value.y },
+				{ scale: withSpring( isPressed.value ? 1.2 : 1 ) },
+			],
+			backgroundColor: isPressed.value ? 'yellow' : 'blue',
+		};
+	} );
+
+	const start = useSharedValue( { x: 0, y: 0 } );
+	const gesture = Gesture.Pan()
+		.onBegin( () => {
+			isPressed.value = true;
+		} )
+		.onUpdate( ( e ) => {
+			offset.value = {
+				x: e.translationX + start.value.x,
+				y: e.translationY + start.value.y,
+			};
+		} )
+		.onEnd( () => {
+			start.value = {
+				x: offset.value.x,
+				y: offset.value.y,
+			};
+		} )
+		.onFinalize( () => {
+			isPressed.value = false;
+		} );
+
+	return (
+		<View style={ ballStyles.container }>
+			<GestureDetector gesture={ gesture }>
+				<Animated.View style={ [ ballStyles.ball, animatedStyles ] } />
+			</GestureDetector>
+		</View>
+	);
+};
+
 export default class VisualEditor extends Component {
 	constructor( props ) {
 		super( props );
@@ -58,11 +123,14 @@ export default class VisualEditor extends Component {
 		const { isAutoScrollEnabled } = this.state;
 
 		return (
-			<BlockList
-				header={ this.renderHeader }
-				safeAreaBottomInset={ safeAreaBottomInset }
-				autoScroll={ isAutoScrollEnabled }
-			/>
+			<>
+				<TestComponent />
+				<BlockList
+					header={ this.renderHeader }
+					safeAreaBottomInset={ safeAreaBottomInset }
+					autoScroll={ isAutoScrollEnabled }
+				/>
+			</>
 		);
 	}
 }
```

</details>

2. Open any post/page.
3. Observe that a blue ball is displayed in the right lower corner.
4. Observe that the ball can be dragged and dropped around the screen.
5. Enable the debugger (use Chrome debugger) from the React Native developer menu.
6. Observe that the app doesn't crash in both Android and iOS when enabling the Chrome debugger.
**NOTE:** When the debugger is connected, Reanimated worklets are not executed, so it's expected that the blue ball can't be dragged when debugging.

## Screenshots or screencast 
N/A

### CODE REVIEW

File: packages/react-native-editor/package.json [hunk 1]
```diff
@@ -65,7 +65,7 @@
 		"react-native-linear-gradient": "https://raw.githubusercontent.com/wordpress-mobile/react-native-linear-gradient/v2.5.6-wp-2/react-native-linear-gradient-2.5.6-wp-2.tgz",
 		"react-native-modal": "^11.10.0",
 		"react-native-prompt-android": "https://raw.githubusercontent.com/wordpress-mobile/react-native-prompt-android/v1.0.0-wp-2/react-native-prompt-android-1.0.0-wp-2.tgz",
-		"react-native-reanimated": "https://raw.githubusercontent.com/wordpress-mobile/react-native-reanimated/2.2.4-wp-1/react-native-reanimated-2.2.4-wp-1.tgz",
+		"react-native-reanimated": "https://raw.githubusercontent.com/wordpress-mobile/react-native-reanimated/update-version-2.4.1/react-native-reanimated-2.4.1-wp-1.tgz",
```
Comment (fluiddot) [inline packages/react-native-editor/package.json]: We have to update the reference once https://github.com/wordpress-mobile/react-native-reanimated/pull/13 is merged and a tag is created.

**Review by geriux — APPROVED** [review]
LGTM! I tested this on both iOS and Android, the Chrome debugger works for both platforms as well as the Reanimated integration 🚀

**Comment by fluiddot** [comment]
@geriux I bumped into a building issue when trying to generate the Android installable build in WordPress-Android ([reference](https://github.com/wordpress-mobile/WordPress-Android/pull/16106#issuecomment-1068129702)). I'm working on a fix but it will imply bumping the `react-native-gesture-handler` version 😭. I'll let you know when the changes are ready, in case you could take another look 🙇.

**Comment by fluiddot** [comment]
> @geriux I bumped into a building issue when trying to generate the Android installable build in WordPress-Android ([reference](https://github.com/wordpress-mobile/WordPress-Android/pull/16106#issuecomment-1068129702)). I'm working on a fix but it will imply bumping the `react-native-gesture-handler` version 😭. I'll let you know when the changes are ready, in case you could take another look 🙇.

This issue will be fixed in https://github.com/wordpress-mobile/react-native-gesture-handler/pull/18. @geriux I assigned you as a reviewer there, let me know if you will be able to review it, thanks 🙇 .

**Comment by geriux** [comment]
> > @geriux I bumped into a building issue when trying to generate the Android installable build in WordPress-Android ([reference](https://github.com/wordpress-mobile/WordPress-Android/pull/16106#issuecomment-1068129702)). I'm working on a fix but it will imply bumping the `react-native-gesture-handler` version 😭. I'll let you know when the changes are ready, in case you could take another look 🙇.
> 
> This issue will be fixed in [wordpress-mobile/react-native-gesture-handler#18](https://github.com/wordpress-mobile/react-native-gesture-handler/pull/18). @geriux I assigned you as a reviewer there, let me know if you will be able to review it, thanks 🙇 .

Thanks for handling that issue @fluiddot! I've just approved the PR 🙇

**Pull request merged**